### ISSUE TITLE
section[["station",i]] has faulty metadata$station

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1908]
Demo below. (Discovered whilst testing for #1907, and the fix will be in a commit that fixes both.)

```R
> library(oce)
> (section[["station",100]]@metadata$stn)
  [1] NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA   
 [12] NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA   
 [23] NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA   
 [34] NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA   
 [45] NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA   
 [56] NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA   
 [67] NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA   
 [78] NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA   
 [89] NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA   
[100] "109"
```

**Comment by dankelley** [comment]
This is not a problem with `[[,section-method`; the error is in `read.section()` or something it calls.  Evidence is below.  Notice that 100, the length of `stn`, is *not* the number of CTD profiles in this section file.
```R
> section@data$station[[100]]@metadata$stn
  [1] NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA
 [13] NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA
 [25] NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA
 [37] NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA
 [49] NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA
 [61] NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA
 [73] NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA
 [85] NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA    NA
 [97] NA    NA    NA    "109"   
```

**Comment by dankelley** [comment]
Fixed in commit c2e8e341e77d36b56622f296a6a25def111c2b9d of the "develop" branch.
